### DISCUSSION TITLE
where can i find instruction to build a custom template?

### DISCUSSION THREAD

**Discussion by nutpantz** [gsantner/markor#1851] [General] [answered]
I would like to build some custom templates for different kinds of journal entries.
But i need to know how to set the formatting to auto fill various fields like date and time in the file name (similar to how 202209192133-zettlekasten.md does it but i want 20220919doctor or 20220919medicine or 20220919journal). And how to auto fill date and time in the template itself.
Is there a manual someplace?

**Comment by gsantner** [comment]
Read https://gsantner.net/blog/2022/07/16/markor-v2.10-custom-file-templates-share-into-remove-tracking-parameters.html

**Reply by nutpantz** [reply]
Maybe I'm blind, but i could not find any formatting information for template files in that link.  Eq: today's date displayed

**Reply by gsantner** [reply]
You are not blind, there is no special formatting or something **for custom** files. You write predefined files and they are copied to the new destination. If you want to insert todays date in whatever format, you have the format button in the bottombar to insert it in that format you need at the time.

If you want more -- please go ahead, start contributing code to the project. Thank you. And - not everything needs to be customizable to the deepest, not everything needs neccassary to be a option.

**Comment by harshad1** [comment]
Custom snippets and templates support 2 special features:

1. Date formats in backticks are interpolated (using java's simpledateformat notation)
2. Your cursor will be placed at the location of the string `%%PLACE_CURSOR_HERE%%`, if it is found in your template.

This isn't documented, unfortunately. I implemented them for my own convenience, but have not taken the time to document them.

If you are able to document these with examples, IMO that would be a pretty great contribution to markor

**Comment by harshad1** [comment, marked as answer]
For example, here is my daily planning template:

```markdown
# `yyyy-MM-dd, EEEE`

**Single priority**: %%PLACE_CURSOR_HERE%%

## Current projects
1. 

## Tasks
- [ ] 

## Plan
### A
- 

## Notes
```

**Reply by steko** [reply]
Thank you for giving this detailed reply about the undocumented feature. However I can't make it work as a template, the date interpolation only works when I insert it as a snippet, is this intended?

**Comment by harshad1** [comment]
It is a bug / oversight. I have a fix for this ready. Not sure when it will make it into a release.

**Comment by ukinaa** [comment]
Perhaps related to the topic, is it possible to have a custom snippet that wraps around selected text, like with built-in ActionButtons (e.g. bold puts ** around a selection)? Currently, this is not the case: if I first select some text and then select my snippet, the text gets replaced by the snippet.

**Reply by harshad1** [reply]
Yeah this is totally feasible. We can add a {{sell}} keyword which gets replaced with the selected text in the snippet.

Could you create a feature request and @ me in it. 

I would like to research how other snippet systems do this too. I generally want markor to be compatible with other apps.

**Reply by ukinaa** [reply]
Thank you, @harshad1, for considering this. I've created a new feature request on this #2059 .
